datasets/ONS-REUK-Service/main.py

```diff
@@ -12,65 +39,126 @@ def right(s, amount):
 def mid(s, offset, amount):
     return s[offset:offset+amount]
 
-import json
 
-scraper = Scraper(json.load(open('info.json'))['landingPage'])
+# In[23]:
+
+
+with open ('info.json') as file:
+    info = json.load(file)
+    
+landingPage = info['landingPage']
+landingPage
+
+
+# In[24]:
+
+
+scraper = Scraper(landingPage)
+scraper.dataset.family = info['families']
 scraper
 
 
+# %%
+
+# In[25]:
+
+
 # %%
 distribution = scraper.distribution(latest=True)
-tabs = distribution.as_databaker()
 distribution
 
+
+# In[26]:
+
+
+tabs = distribution.as_databaker()
+
+
 # %%
+
+# In[27]:
+
+
+trace = TransformTrace()
+datacube_name = info['title']
+columns = ['Functional category', 'Area', 'Value', 'Unit', 'Measure Type', 'Flow', 'Year']
+
 tidied_sheets = []
 
-for i in tabs:
-    if not i.name in ['NUTS1 by category 2011','NUTS1 by category 2012','NUTS1 by category 2013','NUTS1 by category 2014','NUTS1 by category 2015','NUTS1 by category 2016']:
-        continue
-    tab = distribution.as_pandas(sheet_name = i.name)
-    
-    observations = tab.iloc[2:17]
-    observations.rename(columns = observations.iloc[0], inplace = True)
-    observations.drop(observations.index[0], inplace = True)
-    
-    table = pd.melt(observations, id_vars = ['Functional category'], var_name = 'Area', value_name = 'Value')
-    table.Value.dropna(inplace = True)
-    table['Unit'] = 'gbp-million'
-    table['Measure Type'] = 'GBP Total'
-    table['Year']  = right(i.name,4)
-    table['Flow'] = 'exports'    
-    
-    tidied_sheets.append(table)    
-    
-tabNI = tab = distribution.as_pandas(sheet_name = 'Northern Ireland')  
-observations = tabNI.iloc[3:18, :6]
-observations.rename(columns= observations.iloc[0], inplace=True)
-observations.drop(observations.index[0], inplace = True)
-observations.columns.values[0] = 1
-table = pd.melt(observations, id_vars=[1], var_name='Year', value_name='Value')
-table.Value.dropna(inplace =True)
-table.columns = ['Functional category' if x== 1 else x for x in table.columns]
-table['Unit'] = 'gbp-million'
-table['Measure Type'] = 'GBP Total'
-table['Area'] = 'Northern Ireland'
-table['Flow'] = 'exports'
-table['Year'] = table['Year'].apply(lambda x: pd.to_numeric(x, downcast='integer'))
-table['Year'] = table['Year'].astype(int)
-table['Functional category'] = table['Functional category'].map(lambda x: { 
-                    'Total in all categories':'Total in all categories for NUTS1 area' }.get(x, x))
-tidied_sheets.append(table)
+for tab in tabs:
+    if 'NUTS1 by category' in tab.name:
+        print (tab.name)
+        
+        trace.start(datacube_name, tab, columns, scraper.distributions[0].downloadURL)
+        tab_df = distribution.as_pandas(sheet_name = tab.name)
+        
+        trace.obs("Values from the dataframe")
+        observations = tab_df.iloc[2:17]
```

Comment:
I know this wasn't you (and don't worry about changing it in this instance) but try and avoid these kinds of explicit index references in your own work, they tend to be very fragile.

Reply:
All changes are well acknowledged and noted, and will be effected shortly. Thanks so much.